src/Microsoft.ApplicationInsights.AspNetCore/Implementation/ApplicationInsightsInitializer.cs

```diff
@@ -48,11 +49,20 @@ public void Start()
         /// <inheritdoc />
         void IObserver<DiagnosticListener>.OnNext(DiagnosticListener value)
         {
-            foreach (var applicationInsightDiagnosticListener in this.diagnosticListeners)
+            lock (this.subscriptions)
             {
-                if (applicationInsightDiagnosticListener.ListenerName == value.Name)
+                if (shuttingDown)
                 {
-                    this.subscriptions.Add(value.SubscribeWithAdapter(applicationInsightDiagnosticListener));
+                    value.Dispose();
```

Comment:
Q: Should this Dispose call be here?  

If we are the "owners" of `value` then I think it should be disposed, but is it possible that this value is shared?

Reply:
I'll fix this, thanks for answering!